### PR TITLE
No returns required journey acceptance tests

### PR DESCRIPTION
**Pull request by rvsiyad**
https://eaflood.atlassian.net/browse/WATER-4484

As the return requirements journey is now nearing completion, we are starting to create automated acceptance tests for the returns setup functionality.

This also a chance for team members with no experience with [cypress](https://docs.cypress.io/guides/overview/why-cypress) to get to grips with the framework and the [water-abstraction-acceptance-tests](https://github.com/DEFRA/water-abstraction-acceptance-tests).

This PR is primarily focused on creating the acceptance tests for the no-returns required journey. The PR includes test for changing the start date/reason as well as adding, removing and deleting a note.

### CODE REVIEW

File: cypress/e2e/internal/returns-requirements/no-returns-required.cy.js [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+    // confirm we are on the why no returns required page
+    cy.get('.govuk-fieldset__heading').contains('Why are no returns required?')
+
+    // choose returns exception and click continue
+    cy.get('#reason-3').check()
+    cy.contains('Continue').click()
+
+    // confirm we are back on check page and see the reason changes
+    cy.get('.govuk-heading-xl').contains('Check the return requirements for Mr J J Testerson')
+    cy.get('[data-test="reason"]').contains('Transfer licence')
```
Comment (Cruikshanks) [inline cypress/e2e/internal/returns-requirements/no-returns-required.cy.js]: We covered changing the start date and reason on the returns-required journey. I'm happy to be overruled on the reason; it is going to a different page. But changing the start date here is just a duplication of something we are already testing in another test so should be dropped.

**Review by Cruikshanks — CHANGES_REQUESTED** [review]
I think we have a bit of test duplication. But other than that looks great!

File: cypress/e2e/internal/returns-requirements/no-returns-required.cy.js (hunk 1 above)
Comment (rvsiyad) [reply cypress/e2e/internal/returns-requirements/no-returns-required.cy.js]: Removing the start date change tests but I will leave the reason page. Seems the best route

File: cypress/e2e/internal/returns-requirements/no-returns-required.cy.js [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    cy.get('#note').type('This is a note for a no returns requirement.')
+    cy.contains('Confirm').click()
+
+    // confirm we are back on the check page
+    cy.get('.govuk-heading-xl').contains('Check the return requirements')
+
+    // confirm we see pop up notification confirming changes have been made
+    cy.get('.govuk-notification-banner').contains('Changes made')
+
+    // confirm we see the node added
```
Comment (Cruikshanks) [inline cypress/e2e/internal/returns-requirements/no-returns-required.cy.js]:
```suggestion
    // confirm we see the note added
```

File: cypress/e2e/internal/returns-requirements/no-returns-required.cy.js [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    cy.get('#note').type('This is new and improved note for a no return requirement.')
+    cy.contains('Confirm').click()
+
+    // confirm we are back on the check page
+    cy.get('.govuk-heading-xl').contains('Check the return requirements')
+
+    // confirm we see pop up notification confirming changes have been made
+    cy.get('.govuk-notification-banner').contains('Changes made')
+
+    // confirm we see the node added
```
Comment (Cruikshanks) [inline cypress/e2e/internal/returns-requirements/no-returns-required.cy.js]:
```suggestion
    // confirm we see the note added
```

**Review by Cruikshanks — CHANGES_REQUESTED** [review]
Just a typo!

**Review by Cruikshanks — APPROVED** [review]

**Pull request merged**